src/module.c

```diff
@@ -704,10 +703,26 @@ int RM_IsKeysPositionRequest(RedisModuleCtx *ctx) {
  *  keys are at fixed positions. This interface is only used for commands
  *  with a more complex structure. */
 void RM_KeyAtPos(RedisModuleCtx *ctx, int pos) {
-    if (!(ctx->flags & REDISMODULE_CTX_KEYS_POS_REQUEST)) return;
+    if (!(ctx->flags & REDISMODULE_CTX_KEYS_POS_REQUEST) || !ctx->keys_result) return;
     if (pos <= 0) return;
-    ctx->keys_pos = zrealloc(ctx->keys_pos,sizeof(int)*(ctx->keys_count+1));
-    ctx->keys_pos[ctx->keys_count++] = pos;
+
+    getKeysResult *res = ctx->keys_result;
+
+    /* Check overflow */
+    if (res->numkeys == res->size) {
+        /* If we're using the preallocated array, reallocate on the heap
+         * and move the data. Otherwise we can just realloc. */
+        int newsize = res->size * 2;
```

Comment:
above a certain size, `*2` may be too much.
we can consider just adding 1000 instead.

Reply:
Capping the doubling is reasonable but I wouldn't want to use intervals of 1000 on extreme cases.